corehq/apps/userreports/tests/test_indicators.py

```diff
@@ -438,3 +447,83 @@ def testDecimal(self):
         self._check_result(indicator, dict(foo=5.5), Decimal(5.5))
         self._check_result(indicator, dict(foo=None), None)
         self._check_result(indicator, dict(foo="banana"), None)
+
+
+class LedgerBalancesIndicatorTest(SimpleTestCase):
+    def setUp(self):
+        self.spec = {
+            "type": "ledger_balances",
+            "column_id": "soh",
+            "display_name": "Stock On Hand",
+            "ledger_section": "soh",
+            "product_codes": ["abc", "def", "ghi"],
+            "case_id_expression": {
+                "type": "property_name",
+                "property_name": "_id"
+            }
+        }
+        self.stock_states = {'abc': 32, 'def': 85, 'ghi': 11}
+
+    @patch.object(LedgerBalancesIndicator, '_get_values_by_product')
+    def test_ledger_balances_indicator(self, get_values_by_product):
+        get_values_by_product.return_value = self.stock_states
+        indicator = IndicatorFactory.from_spec(self.spec)
+
+        doc = {'_id': 'case1'}
+        values = indicator.get_values(doc, EvaluationContext(doc, 0))
+
+        self.assertEqual(
+            [(val.column.id, val.value) for val in values],
+            [('soh_abc', 32), ('soh_def', 85), ('soh_ghi', 11)]
+        )
+
+
+class TestGetValuesByProduct(TestCase):
```

Comment:
I mocked out the `_get_values_by_product` query in the indicator test, so this tests that explicitly.  It's mostly tedious setup work.